File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.js
var largestInteger = function (nums, k)
{
    const cnt = Array(51).fill(0);

    for (const x of nums)
    {
        cnt[x]++;
    }

    if (k === nums.length)
    {
        return Math.max(...nums);
    }

    if (k === 1)
    {
        let ans = -1;

        for (const x of nums)
        {
            if (cnt[x] === 1)
            {
                ans = Math.max(ans, x);
            }
        }

        return ans;
    }

    const a = cnt[nums[0]] === 1 ? nums[0] : -1;
    const b = cnt[nums[nums.length - 1]] === 1
        ? nums[nums.length - 1]
        : -1;

    return Math.max(a, b);
};

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.scala
object Solution {
    def largestInteger(nums: Array[Int], k: Int): Int = {
        val cnt = Array.fill(51)(0)

        nums.foreach(x => cnt(x) += 1)

        if (k == nums.length) {
            return nums.max
        }

        if (k == 1) {
            var ans = -1

            nums.foreach { x =>
                if (cnt(x) == 1) {
                    ans = math.max(ans, x)
                }
            }

            return ans
        }

        val a = if (cnt(nums.head) == 1) nums.head else -1
        val b = if (cnt(nums.last) == 1) nums.last else -1

        math.max(a, b)
    }
}